File: tutor-backend/src/app/api/schemas/ai.py
from typing import Any, Dict, List
from pydantic import BaseModel


class RawOllamaRequest(BaseModel):
    model: str
    response_format: Dict[str, Any]
    messages: List[Dict[str, Any]]


class AIExerciseOut(BaseModel):
    id: int
    tema: str
    enunciado: str
    dificultad: str
    tipo: str
    explicacion: str | None = None

File: tutor-frontend/src/types/ai.ts
export interface AIRequest {
    model: string;
    response_format: { type: "json_object" };
    messages: { role: "system" | "user" | "assistant"; content: string }[];
  }
  
  export interface AIExerciseOut {
    id: number;
    tema: string;
    enunciado: string;
    dificultad: string;
    tipo: string;
    explicacion: string;
  }
  
  export interface AnswerOut {
    correcto: boolean;
  }
  